### Задание 1

In [13]:
# 1.3
!grep -v '>' sequence.fasta | tr -d '\n' | wc -c
!splitter -sequence sequence.fasta -outseq res.split -size=281236

2812357


Split sequence(s) into smaller sequences


In [14]:
# 1.4
!compseq -sequence sequence.fasta -word 4 -outfile res4.comp
!cat res4.comp | grep AAAA
!cat res4.comp | grep ATAT 
!cat res4.comp | grep ATTA
!cat res4.comp | grep AATT 

Calculate the composition of unique words in sequences
AAAA	42791		0.0152154	0.0039062	3.8951341
ATAT	20764		0.0073831	0.0039062	1.8900835
ATTA	14177		0.0050410	0.0039062	1.2904890
AATT	16433		0.0058431	0.0039062	1.4958458


In [15]:
# 1.5
!shuffleseq -sequence res.split -outseq res.shuffle
!grep -v '>' res.shuffle | tr -d '\n' | wc -c

Shuffle a set of sequences maintaining composition
2812357


In [16]:
# 1.6
!compseq -sequence res.shuffle -word 4 -outfile res4.shuffle.comp
!cat res4.shuffle.comp | grep AAAA 
!cat res4.shuffle.comp | grep ATAT 
!cat res4.shuffle.comp | grep ATTA 
!cat res4.shuffle.comp | grep AATT 

Calculate the composition of unique words in sequences
AAAA	17297		0.0061504	0.0039062	1.5745082
ATAT	16669		0.0059271	0.0039062	1.5173428
ATTA	16772		0.0059637	0.0039062	1.5267186
AATT	16559		0.0058880	0.0039062	1.5073297


In [17]:
# 1.7
# Чтобы посчитать модель Бернулли найдем частоты однобуквенных слов.
# Далее перемножим вероятности букв в нужной последовательности
!compseq -sequence sequence.fasta -word 1 -outfile res1.comp

Calculate the composition of unique words in sequences


| Word | Default | Shuffled | Bernoulli |
| ---- | ------- | -------- | --------- |
| AAAA | 0.015   | 0.006    | 0.006     |
| ATAT | 0.007   | 0.005    | 0.005     |
| ATTA | 0.005   | 0.005    | 0.005     |
| AATT | 0.005   | 0.005    | 0.005     |

### Задание 2

#### 2.1
| UniprotKB | Swiss-Prot | TrEMBL |
| --------- | ---------- | ------ |
| 65902     | 955        | 64947  |

#### 2.2
| UniprotKB | Swiss-Prot | TrEMBL |
| --------- | ---------- | ------ |
| 14355     | 144        | 14211  |

#### 2.3
| UniprotKB | Swiss-Prot | TrEMBL |
| --------- | ---------- | ------ |
| 164       | 29         | 135    | 

#### 2.4
Согласно описанию на uniprot:

```
Function (MYC_HUMAN - P01106)

- Transcription factor that binds DNA in a non-specific manner, yet also specifically recognizes the core sequence 5'-CAC[GA]TG-3' (PubMed:24940000, PubMed:25956029).
- Activates the transcription of growth-related genes (PubMed:24940000, PubMed:25956029).
- Binds to the VEGFA promoter, promoting VEGFA production and subsequent sprouting angiogenesis (PubMed:24940000, PubMed:25956029).
- Regulator of somatic reprogramming, controls self-renewal of embryonic stem cells (By similarity).
- Functions with TAF6L to activate target gene expression through RNA polymerase II pause release (By similarity).
- Positively regulates transcription of HNRNPA1, HNRNPA2 and PTBP1 which in turn regulate splicing of pyruvate kinase PKM by binding repressively to sequences flanking PKM exon 9, inhibiting exon 9 inclusion and resulting in exon 10 inclusion and production of the PKM M2 isoform (PubMed:20010808)
```

То есть, в основном, функция данного белка заключается в регуляции транскрипции большого количества генов. Поиск дополнительной информации показал, что среди мишеней данного белка есть гены, которые учавствуют в апоптозе, регуляции метаболизма и диффиренцировке клеток. 

#### 2.5
Белок Myc принадлежит к семейству факторов транскрипции Myc, которое также включает белки N-Myc и L-Myc.

#### 2.6
| Protein name | 100% identity | 90% identity | 50% identity |
| ------------ | ------------- | ------------ | ------------ |
| MYC_HUMAN    | 6             | 98           | 183          | 